Downsample/Downsample.cpp

```diff
@@ -138,17 +138,26 @@ Eigen::MatrixXf gridSum(const Eigen::MatrixXf &r, int downsamplingFactor)
 
 Eigen::MatrixXf lineSum(const Eigen::MatrixXf &matrix, int downsamplingFactor)
 {
+    Eigen::Map<Eigen::MatrixXf> reshapedMatrixMap(const_cast<float *>(matrix.data()),
+                                                  downsamplingFactor,
+                                                  (int)matrix.rows() * (int)matrix.cols() / downsamplingFactor);
     Eigen::MatrixXf reshapedMatrix =
-        matrix.reshaped(downsamplingFactor, (int)matrix.rows() * (int)matrix.cols() / downsamplingFactor);
+        Eigen::Map<Eigen::MatrixXf>(reshapedMatrixMap.data(), reshapedMatrixMap.rows(), reshapedMatrixMap.cols());
 
     std::function<float(float)> nan_to_zero_functor = nanToZero;
 
     reshapedMatrix = reshapedMatrix.unaryExpr(nan_to_zero_functor);
 
-    return reshapedMatrix.colwise()
-        .sum()
-        .reshaped((int)matrix.rows() / downsamplingFactor, (int)matrix.cols())
-        .transpose();
+    Eigen::MatrixXf reshapedMatrixColwiseSum = reshapedMatrix.colwise().sum();
+
+    Eigen::Map<Eigen::MatrixXf> reshapedMatrixMap2(reshapedMatrixColwiseSum.data(),
```

Comment:
Can you find more descriptive names for `reshapedMatrixMap2` and `reshapedMatrix2`?

Reply:
flattenedMatrix and reshapedMatrix

Reply:
done